plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoClientModule.java

```diff
@@ -46,24 +50,29 @@ public static MongoSession createMongoSession(TypeManager typeManager, MongoClie
     {
         requireNonNull(config, "config is null");
 
-        MongoClientOptions.Builder options = MongoClientOptions.builder();
-
-        options.connectionsPerHost(config.getConnectionsPerHost())
-                .connectTimeout(config.getConnectionTimeout())
-                .socketTimeout(config.getSocketTimeout())
-                .socketKeepAlive(config.getSocketKeepAlive())
-                .sslEnabled(config.getSslEnabled())
-                .maxWaitTime(config.getMaxWaitTime())
-                .maxConnectionIdleTime(config.getMaxConnectionIdleTime())
-                .minConnectionsPerHost(config.getMinConnectionsPerHost())
+        MongoClientSettings.Builder options = MongoClientSettings.builder();
+        options.writeConcern(config.getWriteConcern().getWriteConcern())
                 .readPreference(config.getReadPreference().getReadPreference())
-                .writeConcern(config.getWriteConcern().getWriteConcern());
+                .applyToConnectionPoolSettings(builder -> builder
+                        .maxConnectionIdleTime(config.getMaxConnectionIdleTime(), MILLISECONDS)
+                        .maxWaitTime(config.getMaxWaitTime(), MILLISECONDS)
+                        .minSize(config.getMinConnectionsPerHost())
+                        .maxSize(config.getConnectionsPerHost()))
+                .applyToSocketSettings(builder -> builder.connectTimeout(config.getConnectionTimeout(), MILLISECONDS))
+                .applyToSslSettings(builder -> builder.enabled(config.getSslEnabled()));
 
         if (config.getRequiredReplicaSetName() != null) {
-            options.requiredReplicaSetName(config.getRequiredReplicaSetName());
+            options.applyToClusterSettings(builder -> builder.requiredReplicaSetName(config.getRequiredReplicaSetName()));
+        }
+
+        if (config.getConnectionUrl().isPresent()) {
+            options.applyConnectionString(new ConnectionString(config.getConnectionUrl().get()));
+        }
+        else {
+            options.applyToClusterSettings(builder -> builder.mode(ClusterConnectionMode.SINGLE).hosts(config.getSeeds()));
```

Comment:
If I understand correctly, hard-coded `ClusterConnectionMode.SINGLE` will break the Trino environment that connects to a MongoDB cluster having multiple hosts. 
https://github.com/mongodb/mongo-java-driver/blob/master/driver-core/src/main/com/mongodb/connection/ClusterSettings.java#L585

Reply:
You're right.